cleanup/artifactCleanup/artifactCleanup.groovy

```diff
@@ -78,43 +94,81 @@ executions {
-    log.info "Starting artifact cleanup for repositories $repos, until $months months ago with pacing interval $paceTimeMS ms, dryrun: $dryRun, disablePropertiesSupport: $disablePropertiesSupport"
+if ( configFile.exists() ) {
+  
+    def config = new JsonSlurper().parse(configFile.toURL())
+    log.info "Schedule job policy list: $config.policies"
+
+    config.policies.each{ policySettings ->
+        def cron = policySettings.containsKey("cron") ? policySettings.cron as String : ["0 0 5 ? * 1"]
+        def repos = policySettings.containsKey("repos") ? policySettings.repos as String[] : ["__none__"]
+        def timeUnit = policySettings.containsKey("timeUnit") ? policySettings.timeUnit as String : DEFAULT_TIME_UNIT
+        def timeInterval = policySettings.containsKey("timeInterval") ? policySettings.timeInterval as int : DEFAULT_TIME_INTERVAL
+        def paceTimeMS = policySettings.containsKey("paceTimeMS") ? policySettings.paceTimeMS as int : 0
+        def dryRun = policySettings.containsKey("dryRun") ? new Boolean(policySettings.dryRun) : false
+        def disablePropertiesSupport = policySettings.containsKey("disablePropertiesSupport") ? new Boolean(policySettings.disablePropertiesSupport) : false
+
+        jobs {
+            "scheduledCleanup_$cron"(cron: cron) {
+                log.info "Policy settings for scheduled run at($cron): repo list($repos), timeUnit($timeUnit), timeInterval($timeInterval), paceTimeMS($paceTimeMS) dryrun($dryRun) disablePropertiesSupport($disablePropertiesSupport)"
+                artifactCleanup( timeUnit, timeInterval, repos, log, paceTimeMS, dryRun, disablePropertiesSupport )
+            }
+        }
+    }  
+}
+
+if ( deprecatedConfigFile.exists() && configFile.exists() ) {
+    log.info "The deprecated artifactCleanup.properties and the new artifactCleanup.json are defined in parallel. You should migrate the old file and remove it."
+}
+
+private def artifactCleanup(String timeUnit, int timeInterval, String[] repos, log, paceTimeMS, dryRun = false, disablePropertiesSupport = false) {
+    log.info "Starting artifact cleanup for repositories $repos, until $timeInterval ${timeUnit}s ago with pacing interval $paceTimeMS ms, dryrun: $dryRun, disablePropertiesSupport: $disablePropertiesSupport"
 
     // Create Map(repo, paths) of skiped paths (or others properties supported in future ...)
     def skip = [:]
     if ( ! disablePropertiesSupport && repos){
         skip = getSkippedPaths(repos)
     }
 
-    def monthsUntil = Calendar.getInstance()
-    monthsUntil.add(Calendar.MONTH, -months)
+    def calendarUntil = Calendar.getInstance()
+
+    calendarUntil.add(mapTimeUnitToCalendar(timeUnit), -timeInterval)
+
+    def calendarUntilFormatted = new SimpleDateFormat("yyyy/MM/dd HH:mm").format(calendarUntil.getTime());
+    log.info "Removing all artifacts older than $calendarUntilFormatted"
```

Comment:
This log message should be `"Removing all artifacts now downloaded since $calendarUntilFormatted"`

Reply:
Fixed! tks @DarthFennec